airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/paginators/strategies/offset_increment.py

```diff
@@ -3,27 +3,49 @@
 #
 
 from dataclasses import InitVar, dataclass
-from typing import Any, List, Mapping, Optional
+from typing import Any, List, Mapping, Optional, Union
 
 import requests
 from airbyte_cdk.sources.declarative.requesters.paginators.strategies.pagination_strategy import PaginationStrategy
+from airbyte_cdk.sources.declarative.interpolation import InterpolatedString
+from airbyte_cdk.sources.declarative.types import Config
 from dataclasses_jsonschema import JsonSchemaMixin
 
 
 @dataclass
 class OffsetIncrement(PaginationStrategy, JsonSchemaMixin):
     """
     Pagination strategy that returns the number of records reads so far and returns it as the next page token
+    Examples:
+        # page_size to be a constant integer value
+        pagination_strategy:
+          type: OffsetIncrement
+          page_size: 2
+
+        # page_size to be a constant string value
+        pagination_strategy:
+          type: OffsetIncrement
+          page_size: "2"
+
+        # page_size to be an interpolated string value
+        pagination_strategy:
+          type: OffsetIncrement
+          page_size: "{{ options['items_per_page'] }}"
 
     Attributes:
-        page_size (int): the number of records to request
+        page_size (InterpolatedString): the number of records to request
     """
 
-    page_size: int
+    config: Config
+    page_size: Union[InterpolatedString, str, int]
     options: InitVar[Mapping[str, Any]]
 
     def __post_init__(self, options: Mapping[str, Any]):
         self._offset = 0
+        # InterpolatedString page_size may contain one of int/str types,
+        # so we need to ensure that its `.string` attribute is of *string* type
+        self.page_size.string = str(self.page_size.string)
+        self.page_size = self.page_size.eval(self.config)
```

Comment:
When possible, we try to perform the interpolation evluation `eval()` at runtime instead of parse time in `post_init`. So in this case, self.page_size would be the interpolated string (as you mentioned above) but we should move evaluation of it into the `get_page_size()`.

Now that this could be an interpolated string or just a normal string, the type hints in the getter method `get_page_size() -> Optional[int]` is now a bit misleading. I think we should also add some error handling in the event that this does not evaluate into an integer since it won't behave correctly when `get_page_size()` gets invoked by parent classes.

Reply:
Fixed